e2e/tests/container_create.go

```diff
@@ -432,6 +432,171 @@ func ContainerCreate(opt *option.Option) {
 			// verify log path exists
 			Expect(inspect[0].LogPath).ShouldNot(BeNil())
 		})
+
+		It("should create a container with specified CPU qouta and period options", func() {
+			// define options
+			options.Cmd = []string{"sleep", "Infinity"}
+			options.HostConfig.CPUQuota = 11111
+			options.HostConfig.CPUShares = 2048
+			options.HostConfig.CPUPeriod = 100000
+
+			// create container
+			statusCode, ctr := createContainer(uClient, url, testContainerName, options)
+			Expect(statusCode).Should(Equal(http.StatusCreated))
+			Expect(ctr.ID).ShouldNot(BeEmpty())
+
+			nativeResp := command.Stdout(opt, "inspect", "--mode=native", testContainerName)
+			var nativeInspect []map[string]interface{}
+			err := json.Unmarshal(nativeResp, &nativeInspect)
+			Expect(err).Should(BeNil())
+			Expect(nativeInspect).Should(HaveLen(1))
+
+			// Navigate to the CPU quota value
+			spec, ok := nativeInspect[0]["Spec"].(map[string]interface{})
```

Comment:
Unmarshalling the return value into the [expected interface](https://github.com/containerd/nerdctl/blob/release/1.7/pkg/inspecttypes/native/container.go#L26-L32) should be simpler than casting every value:

requires this import:
```go
nativeinspect "github.com/containerd/nerdctl/pkg/inspecttypes/native"
```

```suggestion
                        nativeResp := command.Stdout(opt, "inspect", "--mode=native", testContainerName)
			var nativeInspect nativeinspect.Container
			err := json.Unmarshal(nativeResp, &nativeInspect)
			Expect(err).Should(BeNil())
			Expect(nativeInspect.Spec).ShouldNot(BeNil())
```

Reply:
The Spec field describes an empty interface, Unmarshalling into this interface is not going to make this simpler as we'll have to continue to cast the remaining KV pairs within the Spec. 
ex 
```
var nativeInspect []nativeinspect.Container
err := json.Unmarshal(nativeResp, &nativeInspect)
Expect(err).Should(BeNil())
Expect(nativeInspect).Should(HaveLen(1))

// Navigate to the process value
spec, ok := nativeInspect[0].Spec.(map[string]interface{})
Expect(ok).Should(BeTrue())
```